src/engine/enginebuffer.cpp

```diff
@@ -789,49 +794,52 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
 
-            // TODO: setting pitchRatio = speed ensures that the "offlinear"
-            // test below will not succeed.  We should find a better way
-            // to ensure that the keylock scaler is not used.
+            // Force pitchRatio to the linear pitch set by speed
             pitchRatio = speed;
-            keylock_enabled = false;
             // This is for the natural speed pitch found on turn tables
         } else if (fabs(speed) < 0.1 && m_pKeylockEngine->get() == RUBBERBAND) {
             // At very slow speeds, Rubberband performs memory allocations which
             // can cause underruns.  Disable keylock under these conditions.
-            pitchRatio = speed;
-            keylock_enabled = false;
-        } else if (!keylock_enabled) {
-            // We might have have temporary speed change, so adjust pitch if not locked
-            // Note: This will not update key and tempo widgets
-            if (tempoRatio) {
-                pitchRatio *= (speed/tempoRatio);
-            }
-        }
 
-        // If either keylock is enabled or the pitch is tweaked we
-        // need to use pitch and time scaling.
-        // Note: we have still click issue when changing the scaler
-
-        // const double kLinearScalerElipsis = 1.00058;
-
-        bool useIndependentPitchAndTempoScaling = false;
-        if (keylock_enabled) {
+            // Force pitchRatio to the linear pitch set by speed
+            pitchRatio = speed;
+        } else if (keylock_enabled) {
             // always use IndependentPitchAndTempoScaling
             // to avoid clicks when crossing the linear pitch
             // in this case it is most likely that the user
             // will have an non linear pitch
+            // Note: We have undesired noise when cossfading between scalers
             useIndependentPitchAndTempoScaling = true;
-        } else if (speed) {
-            double offlinear = pitchRatio / speed;
-            if (offlinear > kLinearScalerElipsis ||
-                    offlinear < 1 / kLinearScalerElipsis) {
-                // offlinear > 1 cent
-                // everything below is not hear-able
-                useIndependentPitchAndTempoScaling = true;
+        } else {
+            // We might have have temporary speed change, so adjust pitch if not locked
+            // Note: This will not update key and tempo widgets
+            if (tempoRatio) {
+                pitchRatio *= (speed / tempoRatio);
             }
+
+            // Check if we are off-linear (the pitch is tweaked) to enable
+            // the keylock scaler even though keylock is disabled
+            if (speed != 0.0) {
+                double offlinear = pitchRatio / speed;
+                if (offlinear > kLinearScalerElipsis ||
+                        offlinear < 1 / kLinearScalerElipsis) {
+                    // off-linear > 1 cent
+                    // everything below is not hear-able
```

Comment:
replace with:  "// only enable keylock scaler if pitch adjustment is at least 1 cent"